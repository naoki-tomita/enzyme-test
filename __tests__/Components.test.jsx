const { mount } = require("enzyme");
const { renderToJson } = require("enzyme-to-json");
const React = require("react");

const { Component, Model } = require("../Component");

describe("test1", () => {
  it("should get element.", async () => {
    const model = new Model();
    const wrapper = mount(
      <Component model={model}/>
    );
    wrapper.find(".up").simulate("click"); // 1
    wrapper.find(".up").simulate("click"); // 2
    wrapper.find(".up").simulate("click"); // 3
    wrapper.find(".class-1-list").childAt(2).find(".class-list__item--text").simulate("click");
    expect(renderToJson(wrapper.render())).toMatchSnapshot();
    expect(wrapper.find(".class-1-list").childAt(2).html().includes("class-selected")).toBe(true);
    await new Promise(r => {
      model.onUpdate(() => {
        if (model.getCount() === 7) {
          r();
        }
      });
    });
    console.log(wrapper.html());
    console.log(wrapper.debug());
    expect(wrapper.find(".class-1-list").children().length).toBe(7);
  });
});