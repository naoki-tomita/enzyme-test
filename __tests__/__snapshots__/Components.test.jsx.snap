// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test1 should get element. 1`] = `
<div>
  <button
    class="up"
  >
    Count up
  </button>
  <button
    class="down"
  >
    Count down
  </button>
  <div
    class="count"
  >
    4
  </div>
  <div
    class="class-1"
  >
    <ul
      class="class-1-list"
    >
      <li
        class="class-list__item"
      >
        <a />
        <a
          class="class-list__item--text"
        >
          ITEM_0
        </a>
      </li>
      <li
        class="class-list__item"
      >
        <a />
        <a
          class="class-list__item--text"
        >
          ITEM_1
        </a>
      </li>
      <li
        class="class-list__item class-selected"
      >
        <a />
        <a
          class="class-list__item--text"
        >
          ITEM_2
        </a>
      </li>
      <li
        class="class-list__item"
      >
        <a />
        <a
          class="class-list__item--text"
        >
          ITEM_3
        </a>
      </li>
    </ul>
  </div>
</div>
`;
